### PR TITLE
extmod/nimble: Do not set GAP device name after sync.

### LINKED ISSUE #6297: BLE.active() change gap_name to "ESP32"
May I know if it is the natural behavior of BLE.active() to change gap_name please? After BLE.active() the gap_name will always changed back to "ESP32"

Tested on the latest unstable and stable version.

Thanks

```
>>> import os
>>> os.uname()
(sysname='esp32', nodename='esp32', release='1.12.0', version='v1.12-662-g8da40baa4 on 2020-07-29', machine='ESP32 module (spiram) with ESP32')
>>> import bluetooth
>>> ble=bluetooth.BLE()
>>> ble.config(gap_name="TESTING")
>>> ble.config('gap_name')
b'TESTING'
>>> ble.active(True)
I (2350) BTDM_INIT: BT controller compile version [c1cbe45]
I (6316) system_api: Base MAC address is not set, read default base MAC address from BLK0 of EFUSE
W (6326) phy_init: failed to load RF calibration data (0x1102), falling back to full calibration
I (6476) phy: phy_version: 4180, cb3948e, Sep 12 2019, 16:39:13, 0, 2
True
>>> ble.config('gap_name')
b'ESP32'
>>>
```

### PR DESCRIPTION
**Pull request by DvdGiessen**
In the NimBLE sync callback we used to set the name to the default name; thus that could override a custom name set by the user.

~This saves the custom name so that it can be properly restored when a sync happens.~

This removes that setting of the default name. NimBLE already has a default which we can define compile-time, no runtime initialisation required. And not setting it means that the GAP name will also not be reset if Bluetooth is toggled off and on again with the `bluetooth.active(bool)` method.

Fixes https://github.com/micropython/micropython/issues/6297, https://github.com/micropython/micropython/issues/7220

### CODE REVIEW

File: extmod/nimble/modbluetooth_nimble.c [hunk 1]
```diff
@@ -351,7 +352,11 @@ STATIC void sync_cb(void) {
     }
 
     DEBUG_printf("sync_cb: Setting device name\n");
-    ble_svc_gap_device_name_set(MICROPY_PY_BLUETOOTH_DEFAULT_GAP_NAME);
+    if (nimble_gap_name[0] != '\0') {
+        ble_svc_gap_device_name_set(nimble_gap_name);
+    } else {
+        ble_svc_gap_device_name_set(MICROPY_PY_BLUETOOTH_DEFAULT_GAP_NAME);
+    }
```
Comment (dpgeorge) [inline extmod/nimble/modbluetooth_nimble.c:359]: Would it be better to just not call `ble_svc_gap_device_name_set()` if the name was already set?  I guess this depends on the internals of NimBLE, if it retains the GAP name if it was set before bringing up the stack.

Comment (DvdGiessen) [reply extmod/nimble/modbluetooth_nimble.c]: Forced-pushed an alternate commit implementing this approach.

Comment (dpgeorge) [reply extmod/nimble/modbluetooth_nimble.c]: This looks good, thanks!

**Comment by dpgeorge** [comment]
This is a good fix, and worth merging.

But note that we are considering to eventually make it so that `network.hostname(...)` be used also to set the BLE GAP name.  That way there's a single name to set for the host and that name is used in all external facing network-like interfaces.  Probably `ble.config(gap_name=...)` would still remain as a way to override the GAP name for the BLE, but the default value would be taken from the setting made by `network.hostname(...)`.

**Comment by DvdGiessen** [comment]
> But note that we are considering to eventually make it so that network.hostname(...) be used also to set the BLE GAP name. 
> Probably ble.config(gap_name=...) would still remain as a way to override the GAP name for the BLE, but the default value would be taken from the setting made by network.hostname(...).

That is possible. Note that due to how NimBLE implements `ble_svc_gap_device_name_set()` that function is safe to call regardless of whether Bluetooth is enabled or not (because all it does is memcopy the parameter to a static variable); so it would be fine to call `ble_svc_gap_device_name_set()` or `ble.config(gap_name=...)` from the (future) implementation of `network.hostname(...)` unconditionally. That doesn't conflict with this PR.

**Comment by dpgeorge** [comment]
Sorry, this got lost.  Now merged.

**Pull request merged**